packages/flutter/lib/src/widgets/localized_resources.dart

```diff
@@ -0,0 +1,354 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' show Locale;
+
+import 'package:flutter/foundation.dart';
+
+import 'container.dart';
+import 'framework.dart';
+
+/// Progress for a [LocalizedResourcesDelegate.load] method.
+enum LocalizedResourcesStatus {
+  /// Loading has not been started, [LocalizedResourcesDelegate.resourcesFor] will return null.
+  none,
+
+  /// Loading is underway, [LocalizedResourcesDelegate.resourcesFor] will return null.
+  loading,
+
+  /// Loading is complete, [LocalizedResourcesDelegate.resourcesFor] will return a valid non-null value.
+  loaded,
+}
+
+/// An encapsulation of all of the resources to be loaded by a
+/// [LocalizedResources] widget.
+///
+/// Typical applications have one [LocalizedResources] widget which is
+/// created by the [WidgetsApp] and configured with the app's
+/// `localizedResourcesDelegate` parameter.
+///
+/// See also:
+///
+///  * [DefaultLocalizedresorucesDelegate], a simple concrete version of
+///    this class.
+abstract class LocalizedResourcesDelegate {
+  /// Start loading the resources for `locale`. The returned future completes
+  /// when the resources have been loaded and cached.
+  ///
+  /// It's assumed that the this method will create one or more objects
+  /// each of which contains a collection or related resources (typically
+  /// defined by one method per resources). The objects will be retrieved
+  /// by [Type] with [resourcesFor].
+  Future<Null> load(Locale locale);
```

Comment:
The split described here doesn't really make sense (the delegate is only needed because it's a loader), but the underlying idea is interesting. One way to address this would be to have the Localizations widget take a `Map<Type, Object>` instead of the delegate, and the `load` method on the delegate would return that map instead of storing it internally. Then, you would have the loading be done by the WidgetsApp rather than the Localizations widget itself. That would let you massively simplify the Localizations widget (it wouldn't need the Stateful/Inherited magic).